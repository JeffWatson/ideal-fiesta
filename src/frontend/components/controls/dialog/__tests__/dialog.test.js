import React from 'react';
import { shallow } from 'enzyme';
import Dialog from '../dialog';

describe('frontend/components/dialog component', () => {
  const props = {
    className: 'class',
    onClose: jest.fn(),
    children: (<div>Children</div>),
  };

  it('renders correctly', () => {
    const button = shallow(<Dialog {...props} />);

    expect(button).toMatchSnapshot();
  });

  it('invokes onClick when clicked', () => {
    const button = shallow(<Dialog {...props} />);

    button.find('.close').simulate('click');
    expect(props.onClose.mock.calls.length).toEqual(1);
  });
});
