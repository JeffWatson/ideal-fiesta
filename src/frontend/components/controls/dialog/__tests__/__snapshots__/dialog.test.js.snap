// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frontend/components/dialog component renders correctly 1`] = `
<div
  className="dialog class"
>
  <Button
    className="close"
    disabled={false}
    onClick={[Function]}
    text="close"
  />
  <div>
    Children
  </div>
</div>
`;
